scripts/install_backend_python_libs.py

```diff
@@ -0,0 +1,406 @@
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Installation script for Oppia python backend libraries."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import collections
+import os
+import shutil
+import subprocess
+
+import python_utils
+from scripts import common
+
+import pkg_resources
+
+
+def _normalize_string_that_contains_python_library_name(original_string):
+    """Python library name strings are case insensitive which means that
+    libraries are equivalent even if the casing of the library names are
+    different. This function normalizes all strings containing python library
+    names so that strings with identical libraries have identical library name
+    strings.
+
+    This is for comparing library names with different case and directory names
+    generated from installing libraries with different case.
+
+    Args:
+        original_string: str. The original string to be normalized.
+
+    Returns:
+        str. A normalized string that is all lowercase.
+    """
+    return original_string.lower()
+
+
+def _get_requirements_file_contents():
+    """Returns a dictionary containing all of the required library names with
+    their corresponding version strings listed in the 'requirements.txt' file.
+
+    Returns:
+        dict(str, str). Dictionary with the name of the library as the key
+        and the version string of that library as the value.
+    """
+    requirements_contents = collections.defaultdict()
+    with python_utils.open_file(
+        common.COMPILED_REQUIREMENTS_FILE_PATH, 'r') as f:
```

Comment:
Out of curiosity, why didn't we go with a yaml or json file for the configuration file since that doesn't require us to code our own parser?

Reply:
We have to use the txt file since the pip-compile function uses it to compile a deterministic version, furthermore, the actually outputted requirements.txt file is in a format that is required by pip.